controllers/database_controller.go

```diff
@@ -1092,9 +1182,38 @@ func (r *DatabaseReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			controller.Owns(&pgv2beta1.PerconaPGCluster{})
 		}
 	}
+	controller.Owns(&corev1.Secret{})
+	controller.Watches(
+		&source.Kind{Type: &corev1.Secret{}},
+		handler.EnqueueRequestsFromMapFunc(r.findObjectsForBackupSecretsName),
+		builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+	)
 	return controller.Complete(r)
 }
 
+func (r *DatabaseReconciler) findObjectsForBackupSecretsName(secret client.Object) []reconcile.Request {
+	attachedDatabaseClusters := &dbaasv1.DatabaseClusterList{}
+	listOps := &client.ListOptions{
```

Comment:
It would be great to document this decision using comments for future generations of developers because I remember that we had discussed it some time ago and right now I can't remember what was the reason. This applies in general to the `SetupWithManager` and the method of using predicates